### ISSUE TITLE
Implement existing foodsoft functionality

### ISSUE THREAD

**Issue by wvengen** [foodcoops/foodsoft-shop#2]
What needs to be implemented before we may consider this for replacing existing Foodsoft member ordering pages? This is a list of foodsoft-shop features to get there.
- [x] Unit bar with quantity and tolerance
- [x] Color amounts according to whether member gets amount or not
- [x] Show article info box _\- moved to tooltips_
- [x] Search article (by name or manufacturer)
- [x] Pagination for order_articles
- [x] Show all ordered articles after clicking on top left _Total_
- [x] Show when order(s) will be closed
- [x] Show message during boxfill phase
  - [x] Restrict decreasing amounts during boxfill phase
- [x] Show box icon watermark in _All_ / _Everyone_ column
- [x] Be nice with stock articles
- [x] Get currency from api / global variable
- [x] I18n
- [x] Handle errors
  - [x] Network timeout _\- "NetworkError when attempting to fetch resource."_
  - [x] Server error _\- notification_
  - [x] Validation failed (e.g. invalid number) _\- notification_
  - [x] Minimum account balance reached _\- "Sorry, your account balance is below the minimum of ..."_
  - [x] Invalid access token (e.g. expired) _\- redirect to auth_
  - [x] Order was closed (in the meantime) _\- "Couldn't find GroupOrderArticle with 'id'=..."_
- [x] Implement optimistic CRUD updates, then
  - [x] Postpone updating the filter until the order_articles have been updated
  - [x] Debounce group order article updates
  - [ ] Refactor group_order_articles actions, saga and reducer (single saga)
  - [ ] `takeLatest` on group_order_articles per id
- [x] Check handling of timezones is correct
- [x] Fix searching when typing slow
- [x] ~~Show who in the ordergroup ordered last (per order?)~~ --> _not here, already in dashboard_
- [x] ~~Show available funds/credit somewhere sensible~~ --> _error msg on update when funds lacking_
- [x] ~~Warning icon at available funds / total when minimum balance reached~~ --> _idem_
- [x] Use png sprite for flags (to reduce asset size)
- [x] Check loading cursor on Chrome (and other browsers; works on Firefox now)
- ...

**Comment by wvengen** [comment]
Debouncing group order article updates means: introducing a small delay between amount changes and sending those to the server, so that a user can press `+` multiple times and only send one change at the end).

This helps to offload the server and make the ui snappier. For the numerical input, it's also nice to have some time to edit a number typing multiple characters, and only sending the final number to the server. That would also be solved by the debouncing.

Trying to implement this, I realised that this either needs a per-group_order_article-attribute whether it is currently being updated or not, or full-fledged optimistic CRUD updates in the store; because otherwise you'd type a number, the number would shortly go back to the old value, and update only after it was send to the server; this would be quite strange.

**Comment by wvengen** [comment]
The debouncing was solved in 5f33163 by doing it in the `DeltaInput` component, as well as holding back updates when `group_order_articles` are updating (using React's [`shouldComponentUpdate`](https://facebook.github.io/react/docs/component-specs.html#updating-shouldcomponentupdate)). This leaves an edge-case where any change after debounce but before a response is received back, is lost. The loading-cursor indicates that the system is busy in this period, so it may be just fine.

**Comment by wvengen** [comment]
Current idea is to use [redux-saga](https://redux-saga.js.org/) (instead of [redux-api]), where `takeLatest` would solve this debouncing issues, I think: the result of the last request sent always takes precedence (also solves a searching issue). This redux-based library (that appeared later) is more powerful.

**Comment by wvengen** [comment]
The current Foodsoft Dashboard screen already shows which users have last ordered per supplier, so this isn't strictly necessary in foodsoft-shop.

**Comment by wvengen** [comment]
An error message is already shown when the available funds are lacking. Maybe that's enough for now.

**Comment by wvengen** [comment]
Tested that it also works with stock orders (incl. max quantity).
